system/src/usb_control_request_channel.cpp

```diff
@@ -183,36 +183,18 @@ particle::UsbControlRequestChannel::~UsbControlRequestChannel() {
 
 int particle::UsbControlRequestChannel::allocReplyData(ctrl_request* ctrlReq, size_t size) {
     const auto req = static_cast<Request*>(ctrlReq);
-    if (req->reply_data) {
-        freeReplyData(req);
-    }
-    // Try to allocate the buffer from the pool
-    if (size <= USB_REQUEST_MAX_POOLED_BUFFER_SIZE &&
-            (req->reply_data = (char*)system_pool_alloc(size, nullptr))) {
-        req->flags |= RequestFlag::POOLED_REP_DATA;
-    } else {
-        // Allocate the buffer dynamically
-        req->reply_data = (char*)t_malloc(size);
-        if (!req->reply_data) {
+    if (size > 0) {
```

Comment:
are we not using the system pool any more? what was the reason for using it to begin with?